### PR TITLE
fix(oas): handling header parameters options

### LINKED ISSUE #120: Rely on `style` and `explode` options while serializing headers
Сurrent header value serialization seems incorrect as IMO it should be something [comma-separated](https://swagger.io/docs/specification/serialization/#header)

_Originally posted by @pmstss in https://github.com/NeuraLegion/har-sdk/pull/114#discussion_r795567143_

### PR DESCRIPTION
**Pull request by pmstss**
fixes #120

### CODE REVIEW

**Comment by pmstss** [comment]
wip (todo: oas2); should be rebased after #135 merge

File: packages/oas/src/converter/parts/headers/Oas3HeadersConverter.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        'content-type',
+        Object.keys(requestBody.content)
+      );
+    }
+
+    return [];
+  }
+
+  protected createAcceptHeaders(_pathObj: OpenAPIV3.OperationObject): Header[] {
+    return [];
```
Comment (derevnjuk) [inline packages/oas/src/converter/parts/headers/Oas3HeadersConverter.ts:30]: I suggest you provide an implementation for this method

```suggestion
    const responses = pathObj.responses as OpenAPIV3.ResponsesObject;
    
    if (isObject(responses)) {
     const code = Math.min(...Object.keys(responses))
    
      return this.createHeaders(
        'accept',
        Object.keys(responses[code].content)
      );
    }

    return [];
```

**Comment by pmstss** [comment]
@derevnjuk now rebased)

File: packages/oas/src/converter/parts/headers/Oas2HeadersConveter.ts [hunk 2]
```diff
@@ -17,6 +21,15 @@ export class Oas2HeadersConverter extends HeadersConverter<OpenAPIV2.Document> {
     return this.createHeaders('accept', pathObj.produces);
   }
 
+  protected convertHeaderParam(
+    headerParam: LocationParam<OpenAPIV2.Parameter>
+  ): Header {
+    return {
+      name: headerParam.param.name,
+      value: this.oas2ValueSerializer.serialize(headerParam) as string
+    };
```
Comment (derevnjuk) [inline packages/oas/src/converter/parts/headers/Oas2HeadersConveter.ts:30]: Use `createHeader()` where it is possible

```suggestion
    return this.createHeader(
      headerParam.param.name,
      this.oas2ValueSerializer.serialize(headerParam) as string
    );
```

File: packages/oas/src/converter/parts/headers/Oas3HeadersConverter.ts [hunk 3]
```diff
@@ -22,10 +26,40 @@ export class Oas3HeadersConverter extends HeadersConverter<OpenAPIV3.Document> {
...
+    value
+  }: LocationParam<OpenAPIV3.ParameterObject>): Header {
+    return {
+      name: param.name,
+      value: decodeURIComponent(
+        this.uriTemplator.substitute(`{x${param.explode ? '*' : ''}}`, {
+          x: value
+        })
+      )
+    };
```
Comment (derevnjuk) [inline packages/oas/src/converter/parts/headers/Oas3HeadersConverter.ts:60]: ditto

**Review by derevnjuk — CHANGES_REQUESTED** [review]

**Review by derevnjuk — APPROVED** [review]

**Pull request merged**